### PR TITLE
Moves coil code into new library

### PR DESCRIPTION
**Pull request by fstanis**
#### WHAT
Follow up for #1868, creates a new library that will host coil-specific implementation of `Paintable` and other image-related implementations in the future.

#### WHY
To remove the hard dependency on coil for use-cases where simpler image loading is possible (e.g. apps only using local resources).

#### HOW
By moving `CoilPaintable` and `rememberArtworkColor` into a new `compose-material-coil` library.

#### Checklist :clipboard:
- [X] Add explicit visibility modifier and explicit return types for public declarations
- [X] Run spotless check
- [X] Run tests
- [X] Update metalava's signature text files

### CODE REVIEW

File: compose-material-coil/src/main/java/com/google/android/horologist/compose/material/coil/ArtworkColor.kt [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+import androidx.core.graphics.drawable.toBitmap
+import androidx.palette.graphics.Palette
+import androidx.wear.compose.material.MaterialTheme
+import coil.imageLoader
+import coil.request.ImageRequest
+import com.google.android.horologist.annotations.ExperimentalHorologistApi
+
+@Composable
+@ExperimentalHorologistApi
+public fun rememberArtworkColor(
```
Comment (yschimke) [inline compose-material-coil/src/main/java/com/google/android/horologist/compose/material/coil/ArtworkColor.kt:35]: Should we replace `Any?` with `String` here? 

Make a blanket rule that we avoid Any in composables?

**Review by yschimke — APPROVED** [review]

File: compose-material-coil/src/main/java/com/google/android/horologist/compose/material/coil/ArtworkColor.kt (hunk 1 above)
Comment (fstanis) [reply compose-material-coil/src/main/java/com/google/android/horologist/compose/material/coil/ArtworkColor.kt]: Missed this - fixed!

**Comment by fstanis** [comment]
I went with `images/base` and `images/coil`instead of `compose-material` and `compose-material-coil` in the latest revision, PTAL.

**Review by yschimke — APPROVED** [review]
LGTM - I'll kick the tires once it's landed.

**Pull request merged**